### PR TITLE
Api link multiple

### PR DESCRIPTION
**Pull request by will-moore**
This improves performance of cut, copy & paste and drag 'n' drop actions in jsTree.
It makes a single AJAX call to create or delete multiple links for each action.
In each case, check that child counts are updated correctly and confirm the links have been created / deleted successfully by refreshing the tree.
Also check that the centre panel is correct and in sync with the tree.
Include cases where one or more of the children being moved is already contained in the destination parent.

Tests:
 - Drag and drop images, datasets & plates, moving from one parent to another in each case.
 - Copy and paste, as above.
 - Cut. Select a number of child nodes, some of which are also in other containers, others will become orphaned. Then Paste.
 
Any other actions you can think of?

### CODE REVIEW

**Comment by dominikl** [comment]
Tested drag&drop of images, datasets and plates with firefox - works fine.

**Comment by dominikl** [comment]
Tested copy, paste and cut as well - works fine.

I just noticed a behaviour I wasn't aware of: If I cut a dataset which is contained in one project only, it is moved to the parent folder (same is true for Insight). Guess that's expected, right?

**Comment by sbesson** [comment]
There are still 9 failing integration tests - see https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-integration-web/87/

**Comment by will-moore** [comment]
Fixed 1 failing test - the others were due to https://github.com/openmicroscopy/openmicroscopy/pull/4208

**Comment by pwalczysko** [comment]
The cut test performed on Win7 Chrome.
All went as expected.

File: components/tools/OmeroWeb/test/integration/test_csrf.py [hunk 1]
```diff
@@ -284,19 +281,19 @@ def test_paste_move_remove_deletamany_image(self):
...
         _delete_response(self.django_client, request_url, data)
-        _csrf_delete_response(self.django_client,
-                              request_url,
-                              json.dumps(data),
-                              content_type="application/json")
+        response = _csrf_delete_response(self.django_client,
+                                         request_url,
+                                         json.dumps(data),
+                                         content_type="application/json")
+        # Response will contain any remaing links from image
```
Comment (mtbc) [inline components/tools/OmeroWeb/test/integration/test_csrf.py]: remaining

**Comment by mtbc** [comment]
https://ci.openmicroscopy.org/job/OMERO-DEV-merge-robotframework/ is still failing; is that related? Otherwise, looks good to me!

**Comment by will-moore** [comment]
@mtbc Thanks - No, those robot tests are unrelated to this PR. Fixed comment.

**Comment by mtbc** [comment]
:+1: As functional testing already done, once travis passes then good to merge.

**Comment by will-moore** [comment]
Ticket created https://trac.openmicroscopy.org/ome/ticket/13049 as discussed with @joshmoore. Good to merge now I think.

**Pull request merged**